src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -82,4 +80,84 @@ public void updateMeal(int mealId, Meal meal) {
         System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
     }
 
-}
\ No newline at end of file
+
+    @Override
+    public Map<String, List<Meal>> mealsByCategories() {
+        List<Meal> allMeals = getMeals();
+        Map<String, List<Meal>> mealsByCategoriesList = new HashMap<>();
+        for (Meal meal : allMeals) {
+            mealsByCategoriesList.computeIfAbsent(meal.getCategory(), k -> new ArrayList<>()).add(meal);
+        }
+
+        return mealsByCategoriesList;
+    }
+
+
+    private int randomNumberGenerator(int max) {
+        Random random = new Random();
+        return random.nextInt(max);
+    }
+
+
+    @Override
+    // generate 7 random meals per each day
+    public HashMap<DaysOfTheWeek, Meal> generateWeeklyMeals(HashMap<String, Integer> userInput) {
+        HashMap<DaysOfTheWeek, Meal> weeklyPlan = new HashMap<>();
+        Map<String, List<Meal>> mealsByCategories = mealsByCategories();
+
+        HashMap<String, Integer> request = new HashMap<>(userInput);
```

Comment:
This is technically controller logic. In your controller you have logic which checks if the request is empty you even have a println about it. You should move setting the request to this default into that if check. Put it under this line basically: `System.out.println("Using default values");`